samples/msal-node-samples/auth-code/test/auth-code-b2c.spec.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import puppeteer from "puppeteer";
+import { Screenshot, createFolder, setupCredentials } from "../../../e2eTestUtils/TestUtils";
+import { NodeCacheTestUtils } from "../../../e2eTestUtils/NodeCacheTestUtils";
+import { LabClient } from "../../../e2eTestUtils/LabClient";
+import { LabApiQueryParams } from "../../../e2eTestUtils/LabApiQueryParams";
+import { B2cProviders, UserTypes } from "../../../e2eTestUtils/Constants";
+import {
+    b2cLocalAccountEnterCredentials,
+    SCREENSHOT_BASE_FOLDER_NAME,
+    validateCacheLocation,
+    SAMPLE_HOME_URL
+ } from "../../testUtils";
+
+import { PublicClientApplication } from "../../../../lib/msal-node/dist";
+
+// Set test cache name/location
+const TEST_CACHE_LOCATION = `${__dirname}/data/b2c.cache.json`;
+
+// Get flow-specific routes from sample application
+const getTokenAuthCode = require("../index");
+
+// Build cachePlugin
+const cachePlugin = require("../../cachePlugin.js")(TEST_CACHE_LOCATION);
+
+// Load scenario configuration
+const config = require("../config/B2C.json");
+
+describe("Auth Code B2C PPE Tests", () => {
+    jest.retryTimes(1);
+    jest.setTimeout(45000);
+    let browser: puppeteer.Browser;
+    let context: puppeteer.BrowserContext;
+    let page: puppeteer.Page;
+    let port: string;
+    let homeRoute: string;
+
+    let username: string;
+    let accountPwd: string;
+
+    const screenshotFolder = `${SCREENSHOT_BASE_FOLDER_NAME}/auth-code/b2c`;
+
+    beforeAll(async() => {
+        await validateCacheLocation(TEST_CACHE_LOCATION);
+        // @ts-ignore
+        browser = await global.__BROWSER__;
+        // @ts-ignore
+        port = 3004;
```

Comment:
Where does this port come from?  Please add a comment to the code on why this port number.

Reply:
Remark added